From: John Johnansen <jjohansen@suse.de>
Subject: log audit message type so it is present in syslog
Patch-mainline: no
References: 304491

Audit messages that go to syslog don't contain the audit type, which
the apparmor tools use to classify messages.  This patch logs the audit
type as part of the AppArmor message so it is present in messages sent
to syslog. 

Signed-off-by: John Johansen <jjohansen@suse.de>

---
 security/apparmor/main.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/security/apparmor/main.c
+++ b/security/apparmor/main.c
@@ -326,8 +326,10 @@ static int aa_audit_base(struct aa_profi
 		return type == AUDIT_APPARMOR_ALLOWED ? 0 : -ENOMEM;
 	}
 
+	audit_log_format(ab, " type=%d", type);
+
 	if (sa->operation)
-		audit_log_format(ab, "operation=\"%s\"", sa->operation);
+		audit_log_format(ab, " operation=\"%s\"", sa->operation);
 
 	if (sa->info)
 		audit_log_format(ab, " info=\"%s\"", sa->info);
